snapcraft/tests/plugins/test_rust.py

```diff
@@ -65,6 +65,10 @@ class Options:
         self.addCleanup(patcher.stop)
 
     def test_cross_compile(self):
+        # unset http and https proxies.
+        self.useFixture(fixtures.EnvironmentVariable('http_proxy', None))
+        self.useFixture(fixtures.EnvironmentVariable('https_proxy', None))
```

Comment:
That seems like the wrong approach since it becomes impossible to run the unit test with a proxy...

More importantly, I don't think the unit test needs to download anything. The run_mock ensures rustup and cargo will never be executed - we should just mock download() as well.